internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -166,38 +169,56 @@ func (r *ClusterDeploymentsReconciler) updateIfNeeded(ctx context.Context, clust
 
 	spec := cluster.Spec
 
-	updateString := func(new, old string, target **string) {
+	updateString := func(new, old string, target **string, targetName string) {
```

Comment:
Maybe instead of adding to every target we can print `params` once before we call for update?

Reply:
the issue is that I didn't find any convenient way to format-print this struct which includes pointers, in a way that it will print their values instead of their addresses. any idea?

Reply:
it's a json, so lets marshal it and print no?